src/containers/trends/TrendsContainer.js

```diff
@@ -57,16 +57,20 @@ export class TrendsContainer extends React.Component {
       low: PropTypes.shape({ boundary: PropTypes.number.isRequired }).isRequired,
       'very-low': PropTypes.shape({ boundary: PropTypes.number.isRequired }).isRequired,
     }).isRequired,
-    bgUnits: PropTypes.oneOf(['mg/dL', 'mmol/L']),
-    extentSize: PropTypes.oneOf([7, 14, 28]),
+    bgUnits: PropTypes.oneOf(['mg/dL', 'mmol/L']).isRequired,
+    extentSize: PropTypes.oneOf([7, 14, 28]).isRequired,
     initialDatetimeLocation: PropTypes.string,
     showingSmbg: PropTypes.bool.isRequired,
     showingCbg: PropTypes.bool.isRequired,
     smbgRangeOverlay: PropTypes.bool.isRequired,
     smbgGrouped: PropTypes.bool.isRequired,
     smbgLines: PropTypes.bool.isRequired,
     smbgTrendsComponent: PropTypes.func.isRequired,
-    timePrefs: PropTypes.object.isRequired,
+    timePrefs: PropTypes.shape({
+      timezoneAware: PropTypes.bool.isRequired,
+      timezoneName: PropTypes.string.isRequired,
+    }).isRequired,
+    yScaleClampTop: PropTypes.object.isRequired,
     // data (crossfilter dimensions)
     cbgByDate: PropTypes.object.isRequired,
```

Comment:
These are crossfilter instances with lots of instance methods. I suppose I could at least document the methods we use in the component? Thoughts @krystophv?